src/Services/Air/Air.js

```diff
@@ -134,19 +135,28 @@ module.exports = (settings) => {
     },
 
     ticket(options) {
-      return this.getPNR(options).then((booking) => {
-        const ticketParams = Object.assign({}, options, {
-          ReservationLocator: booking.uapi_reservation_locator,
-        });
-        return service.ticket(ticketParams)
-          .then(result => result, (err) => {
-            if (err instanceof AirRuntimeError.TicketingFoidRequired) {
-              return this.getPNR(options)
-                .then(updatedBooking => service.foid(updatedBooking))
-                .then(() => service.ticket(ticketParams));
-            }
-            return Promise.reject(err);
-          });
+      return retry({ retries: 3 }, (again, number) => {
+        if (settings.debug && number > 1) console.log(`ticket ${options.pnr} retry number ${number}`);
+        return (options.ReservationLocator ? Promise.resolve(options) :
```

Comment:
Please keep following style:

```
predicate
  ? action
  : another action
```

Reply:
Also, as you never extend `options` itself, `options` will never have `ReservationLocator` attribute at this point.

Reply:
It does have `ReservationLocator` on next retry, see the unit tests added for Air.
It happens because of retry call chain.